ZipArchive.cs

```diff
@@ -692,11 +692,40 @@ public ZipEntry ReadEntry (string entryName, bool caseSensitive = false)
 		}
 
 		public ZipEntry ReadEntry (ulong index)
+		{
+			return ReadEntry (index, throwIfDeleted: true);
+		}
+
+		/// <summary>
+		/// Read a zip entry, given an index.
+		/// 
+		/// When throwIfDeleted is true, if the entry is deleted then an exception is thrown (the error will be
+		/// ErrorCode.Deleted or ErrorCode.Inval, depending on whether the deleted entry previously existed in
+		/// the zip or was newly added - that's just how libzip handles that). If throwIfDeleted is false then
+		/// null is returned for deleted entries and an exception is just thrown for other errors.
+		/// </summary>
+		/// <param name="index">index to read</param>
+		/// <param name="throwIfDeleted">whether to return null or throw an exception for deleted entries</param>
+		/// <returns></returns>
+		public ZipEntry ReadEntry (ulong index, bool throwIfDeleted)
```

Comment:
Presumably at some point in the future, we'll want to enable C#8 Nullable Reference Types in this repo.

Should we do so, this method would need to return `ZipEntry?`.

An alternative approach would be to leave this method unchanged as `ZipEntry ReadEntry(long index)`, and add a *new* method which returns a `ZipEntry?`.

Thus, a "new' method instead of a `ReadEntry()` overload.

Question is, what should said new method be named?

Reply:
We leave it with the same name. Overloads don't *need* to have identical return types.